### PR TITLE
Add aspect ratio support to `Image` rule

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Rule/Image/ImageHandler.php [hunk 1]
```diff
@@ -138,4 +142,35 @@ private function getFilePath(mixed $value): ?string
...
+        ValidationContext $context,
+        Result $result,
+    ): void {
+        if ($rule->getAspectRatioWidth() === null || $rule->getAspectRatioHeight() === null) {
+            return;
+        }
+
+        $validatedAspectRatio = $validatedWidth / $validatedHeight;
+        $expectedAspectRatio = $rule->getAspectRatioWidth() / $rule->getAspectRatioHeight();
+        $absoluteMargin = $rule->getAspectRatioMargin() / 100;
```
Comment (arogachev) [inline src/Rule/Image/ImageHandler.php:159]: Fix mutant.

File: tests/Rule/Image/ImageTest.php [hunk 2]
```diff
@@ -107,6 +199,56 @@ public function dataValidationFailed(): array
         ];
     }
 
+    public function dataValidationFailedAspectRatio(): array
+    {
+        return [
+            'default aspect ratio margin , smaller height' => [
+                new ImageInfo(800, 599),
+                new Image(aspectRatioWidth: 4, aspectRatioHeight: 3),
+                ['' => ['The aspect ratio of image "" must be 4:3 with margin 0%.']],
```
Comment (samdark) [inline tests/Rule/Image/ImageTest.php:208]: Should it error with `""`? Isn't it better to omit it in the message in this case?

**Review by samdark — APPROVED** [review]
Overall :+1: Nice feature.

File: tests/Rule/Image/ImageTest.php (hunk 2 above)
Comment (arogachev) [reply tests/Rule/Image/ImageTest.php:208]: It's already handled in #644, but the changes are in the branch for major release.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 - New #665: Add methods `addErrorWithFormatOnly()` and `addErrorWithoutPostProcessing()` to `Result` object (@vjik)
 - Enh #668: Clarify psalm types in `Result` (@vjik)
-- New #670, #680: Add `Image` validation rule (@vjik, @arogachev)
+- New #670, #677, #680: Add `Image` validation rule (@vjik, @arogachev)
```
Comment (vjik) [inline CHANGELOG.md:7]:
```suggestion
- New #670, #677: Add `Image` validation rule (@vjik, @arogachev)
```

File: src/Rule/Image/ImageAspectRatio.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule\Image;
+
+/**
+ * {@link https://en.wikipedia.org/wiki/Aspect_ratio_(image)}
```
Comment (vjik) [inline src/Rule/Image/ImageAspectRatio.php:8]:
```suggestion
 * @link https://en.wikipedia.org/wiki/Aspect_ratio_(image)
```

File: tests/Support/RuleHandlerResolver/ImageHandlerContainer.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Yiisoft\Validator\Tests\Support\RuleHandlerResolver;
+
+use Yiisoft\Validator\Rule\Image\ImageHandler;
+use Yiisoft\Validator\Rule\Image\ImageInfo;
+use Yiisoft\Validator\RuleHandlerInterface;
+use Yiisoft\Validator\RuleHandlerResolverInterface;
+use Yiisoft\Validator\Tests\Rule\Image\StubImageInfoProvider;
+
+final class ImageHandlerContainer implements RuleHandlerResolverInterface
```
Comment (vjik) [inline tests/Support/RuleHandlerResolver/ImageHandlerContainer.php:13]: You can use `SimpleContainer` instead of new class.

**Review by vjik — APPROVED** [review]

**Pull request merged**